### PR TITLE
Enable forwarding multiple properties

### PR DESCRIPTION
**Pull request by dfed**
This PR is bigger than it looks – most of the lines here are added tests.

This change enables a single `@Instantiable` type to have multiple `@Forwarded` properties. This is not a breaking change.

### CODE REVIEW

File: Sources/SafeDICore/Extensions/CollectionExtensions.swift [hunk 1]
```diff
@@ -21,37 +21,51 @@
 import SwiftSyntax
 import SwiftSyntaxBuilder
 
-extension Array where Element == Dependency {
-
-    var removingDuplicateInitializerArguments: Self {
```
Comment (dfed) [inline Sources/SafeDICore/Extensions/CollectionExtensions.swift:26]: This wasn't necessary. I think I wrote this very early on before dramatically modifying the shape of the DI system.

File: Sources/SafeDICore/Models/Property.swift [hunk 2]
```diff
@@ -61,6 +61,14 @@ public struct Property: Codable, Hashable, Comparable, Sendable {
         )
     }
 
+    var asTupleElement: TupleTypeElementSyntax {
+        TupleTypeElementSyntax(
+            firstName: .identifier(label),
+            colon: .colonToken(),
```
Comment (dfed) [inline Sources/SafeDICore/Models/Property.swift:68]: I don't know why we don't need trailing trivia here but... we don't per the tests.

File: Sources/SafeDICore/Models/TypeDescription.swift [hunk 3]
```diff
@@ -44,8 +44,8 @@ public enum TypeDescription: Codable, Hashable, Comparable, Sendable {
     indirect case array(element: TypeDescription)
     /// A dictionary. e.g. [Int: String]
     indirect case dictionary(key: TypeDescription, value: TypeDescription)
-    /// A tuple. e.g. (Int, String)
-    indirect case tuple([TypeDescription])
+    /// A tuple. e.g. (Int, string: String)
+    indirect case tuple([TupleElement])
```
Comment (dfed) [inline Sources/SafeDICore/Models/TypeDescription.swift:48]: Our generated `ForwardedArguments` tuples have labels, so we need our `TypeDescription` to know about labels now.

File: Sources/SafeDICore/Models/TypeDescription.swift [hunk 4]
```diff
@@ -179,9 +192,9 @@ public enum TypeDescription: Codable, Hashable, Comparable, Sendable {
                 // This is a type that is lazily instantiated.
                 // The first generic is the built type.
                 return builtType
-            } else if name == Dependency.forwardingInstantiatorType, let builtType = generics.last {
+            } else if name == Dependency.forwardingInstantiatorType, let builtType = generics.dropFirst().first {
                 // This is a type that is lazily instantiated with forwarded arguments.
-                // The last generic is the built type.
+                // The second generic is the built type.
```
Comment (dfed) [inline Sources/SafeDICore/Models/TypeDescription.swift:197]: no functional change here with today's code. just future-proofing.

File: Sources/SafeDIMacros/Macros/InstantiableMacro.swift [hunk 5]
```diff
@@ -189,20 +189,67 @@ public struct InstantiableMacro: MemberMacro {
...
+                )
+            ]
+        }
+    }
+
     // MARK: - InstantiableError
 
     private enum InstantiableError: Error, CustomStringConvertible {
         case decoratingIncompatibleType
-        case tooManyForwardedProperties
```
Comment (dfed) [inline Sources/SafeDIMacros/Macros/InstantiableMacro.swift:196]: :wave:

File: Sources/SafeDI/DelayedInstantiation/ForwardingInstantiator.swift [hunk 6]
```diff
@@ -27,21 +27,21 @@
 /// - SeeAlso: `Instantiator`
 /// - Note: This class is the sole means for instantiating an `@Instantiable` type with a `@Forwarded`
 ///   property within the SafeDI framework.
```
Comment (dfed) [inline Sources/SafeDI/DelayedInstantiation/ForwardingInstantiator.swift:29]:
```suggestion
/// - Note: This class is the sole means for instantiating an `@Instantiable` type with `@Forwarded`
///   properties within the SafeDI framework.
```

File: Tests/SafeDIToolTests/SafeDIToolTests.swift [hunk 7]
```diff
@@ -629,6 +629,264 @@ final class SafeDIToolTests: XCTestCase {
...
+
+                    @Instantiated(fulfilledByType: "LoggedInViewController")
+                    let loggedInViewControllerBuilder: ForwardingInstantiator<LoggedInViewController.ForwardedArguments, UIViewController>
+
+                    private let derivedValue: Bool
+
+                    func login(username: String, password: String) {
+                        Task { @MainActor in
+                            let user = await authService.login(username: username, password: password)
+                            let loggedInViewController = loggedInViewControllerBuilder.instantiate((user.id, user.name))
```
Comment (dfed) [inline Tests/SafeDIToolTests/SafeDIToolTests.swift:824]:
```suggestion
                            let loggedInViewController = loggedInViewControllerBuilder.instantiate((userID: user.id, userName: user.name))
```

File: Tests/SafeDIToolTests/SafeDIToolTests.swift [hunk 8]
```diff
@@ -629,6 +629,264 @@ final class SafeDIToolTests: XCTestCase {
...
+
+                    @Instantiated(fulfilledByType: "LoggedInViewController")
+                    let loggedInViewControllerBuilder: ForwardingInstantiator<(userID: String, userName: String), UIViewController>
+
+                    private let derivedValue: Bool
+
+                    func login(username: String, password: String) {
+                        Task { @MainActor in
+                            let user = await authService.login(username: username, password: password)
+                            let loggedInViewController = loggedInViewControllerBuilder.instantiate((user.id, user.name))
```
Comment (dfed) [inline Tests/SafeDIToolTests/SafeDIToolTests.swift:695]:
```suggestion
                            let loggedInViewController = loggedInViewControllerBuilder.instantiate((userId: user.id, userName: user.name))
```

**Review by MrAdamBoyd — APPROVED** [review]
We caught everything in the meeting it looks like. LGTM!

**Pull request merged**